lib/manager/bazel/extract.ts

```diff
@@ -53,63 +54,96 @@ function parseUrl(urlString: string): UrlParsedResult | null {
   return null;
 }
 
-function findBalancedParenIndex(longString: string): number {
-  /**
-   * Minimalistic string parser with single task -> find last char in def.
-   * It treats [)] as the last char.
-   * To find needed closing parenthesis we need to increment
-   * nesting depth when parser feeds opening parenthesis
-   * if one opening parenthesis -> 1
-   * if two opening parenthesis -> 2
-   * if two opening and one closing parenthesis -> 1
-   * if ["""] found then ignore all [)] until closing ["""] parsed.
-   * https://github.com/renovatebot/renovate/pull/3459#issuecomment-478249702
-   */
-  let intShouldNotBeOdd = 0; // openClosePythonMultiLineComment
-  let parenNestingDepth = 1;
-  return [...longString].findIndex((char, i, arr) => {
-    switch (char) {
-      case '(':
-        parenNestingDepth++;
-        break;
-      case ')':
-        parenNestingDepth--;
-        break;
-      case '"':
-        if (i > 1 && arr.slice(i - 2, i).every((prev) => char === prev)) {
-          intShouldNotBeOdd++;
-        }
-        break;
-      default:
-        break;
+const dummyLexer = {
+  main: {
+    lineComment: { match: /#.*?$/ },
+    leftParen: { match: '(' },
+    rightParen: { match: ')' },
+    longDoubleQuoted: {
+      match: '"""',
+      push: 'longDoubleQuoted',
+    },
+    doubleQuoted: {
+      match: '"',
+      push: 'doubleQuoted',
+    },
+    def: {
+      match: new RegExp(
+        [
+          'container_pull',
+          'http_archive',
+          'http_file',
+          'go_repository',
+          'git_repository',
+        ].join('|')
+      ),
+    },
+    unknown: { match: /[^]/, lineBreaks: true },
+  },
+  longDoubleQuoted: {
+    stringFinish: { match: '"""', pop: 1 },
+    char: { match: /[^]/, lineBreaks: true },
+  },
+  doubleQuoted: {
+    stringFinish: { match: '"', pop: 1 },
+    char: { match: /[^]/, lineBreaks: true },
+  },
+};
+
+function parseContent(content: string): string[] {
+  const lexer = moo.states(dummyLexer);
+  lexer.reset(content);
+  let balance = 0;
+
+  let def: null | string = null;
+  const result: string[] = [];
+
+  const finishDef = () => {
+    if (def !== null) {
+      result.push(def);
     }
+    def = null;
+  };
 
-    return !parenNestingDepth && !(intShouldNotBeOdd % 2) && char === ')';
-  });
-}
+  const startDef = () => {
+    finishDef();
+    def = '';
+  };
 
-function parseContent(content: string): string[] {
-  return [
-    'container_pull',
-    'http_archive',
-    'http_file',
-    'go_repository',
-    'git_repository',
-  ].reduce(
```

Comment:
This iteration caused so huge diff on rewritten snapshot (note other snapshot tests are same).